ReactWindows/ReactNative.Shared/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,229 @@
+using ReactNative.UIManager;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Automation.Peers;
+
+namespace ReactNative.Common
+{
+    /// <summary>
+    /// Accessibility helper methods.
+    /// </summary>
+    public static class AccessibilityHelper
+    {
+        #region ImportantForAccessibility
+
+        /// <summary>
+        /// Sets the ImportantForAccessibility property for uiElement.
+        /// It uses AutomationProperties.AccessibilityView to implement the element and its children
+        /// visibility to narrator. ImportantForAccessibility value is stored as an attached property.
+        /// </summary>
+        /// <param name="uiElement"></param>
+        /// <param name="importantForAccessibilityNew"></param>
+        public static void SetImportantForAccessibility(UIElement uiElement, ImportantForAccessibility importantForAccessibilityNew)
+        {
+            var importantForAccessibilityOld = GetImportantForAccessibilityAttached(uiElement);
+            if (importantForAccessibilityOld == importantForAccessibilityNew)
+            {
+                return;
+            }
+
+            SetImportantForAccessibilityAttached(uiElement, importantForAccessibilityNew);
+
+            var uiElementAutomationPeer = FrameworkElementAutomationPeer.FromElement(uiElement);
+            // If this element is 'hidden' by a parrent, do not update AccessibilityView property.
+            if (IsHiddenElement(uiElementAutomationPeer))
+            {
+                return;
```

Comment:
>return [](start = 16, length = 6)

so if we first set importantForAccessibility on parent to a hiding one (Yes/NoHideDescendants) and then on a child then the child's setter is noop and looses the value? what happens then if we set on parent to Auto or No - looks like we've lost the child's value then. Looks like a bug.

Reply:
SetImportantForAccessibility will always set the ImportantForAccessibility flag only on the element for which it is invoked.
The HiddenElement will check the parent chain for Yes/NoHideDescendents. If the element is not "hidden", its children's AccessibilityView will be updated properly. If it is "hidden", there will be no effect on AccessibilityView property for the element since the tree is hidden (Raw). If the parent setting changes, the procedure SetChildrenAccessibilityViewFromImportantForAccessibility will be invoked and it will make sure that the element and all of its children have AccessibilityView that correctly matches the value of ImportantForAccessibility

---
In reply to: [163118867](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163118867) [](ancestors = 163118867)

Reply:
Should we add a comment about why we don't need to update AccessibilityView on the descendants in this case?

---
In reply to: [163136326](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163136326) [](ancestors = 163136326,163118867)

Reply:
Added detailed comment with explanation.

---
In reply to: [163339206](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163339206) [](ancestors = 163339206,163136326,163118867)